streamer/src/nonblocking/quic.rs

```diff
@@ -702,12 +703,33 @@ async fn handle_connection(
     );
     let stable_id = connection.stable_id();
     stats.total_connections.fetch_add(1, Ordering::Relaxed);
+    let mut next_interval = tokio::time::Instant::now()
+        .checked_add(Duration::from_millis(100))
+        .unwrap();
+    let mut streams_in_current_interval: u64 = 0;
     while !stream_exit.load(Ordering::Relaxed) {
+        if matches!(peer_type, ConnectionPeerType::Unstaked) {
+            if streams_in_current_interval >= MAX_UNSTAKED_STREAMS_PER_CONNECTION_PER_100MS {
+                tokio::time::sleep_until(next_interval).await;
```

Comment:
if we block here, handshakes are going to back up in quinn and/or the OS buffers. should load shed imo

Reply:
I concur, the sleep is also not free -- the runtime driver has to constantly poll and drive it especially when the sleep interval is small.

Reply:
We could move the throttling to the handle_chunk. read it but do not send down stream

Reply:
I'll try to move it to handle_chunk

Reply:
Dropping the stream after accepting it will let client keep creating more streams, as the limit set by `set_max_concurrent_uni_streams` will not exhaust. So we'll still see more streams coming in.